packages/query/src/tag.ts

```diff
@@ -1,12 +1,26 @@
-import { processTSQueryAST } from './preprocessor-ts';
-import { processSQLQueryIR } from './preprocessor-sql';
 import { QueryIR } from './loader/sql';
 import { parseTSQuery, TSQueryAST } from './loader/typescript';
+import { processSQLQueryIR } from './preprocessor-sql';
+import { processTSQueryAST } from './preprocessor-ts';
 
 export interface IDatabaseConnection {
   query: (query: string, bindings: any[]) => Promise<{ rows: any[] }>;
 }
 
+function mapQueryResultRows(rows: any[]): any[] {
+  for (const row of rows) {
+    for (const column in row) {
+      const lastCharacter = column[column.length - 1];
+      const isHintedColumn = lastCharacter === '!' || lastCharacter === '?';
+      if (isHintedColumn) {
+        row[column.slice(0, -1)] = row[column];
+        delete row[column];
+      }
+    }
+  }
+  return rows;
+}
```

Comment:
As a minor cleanup:
```suggestion
/** Removes column modifier suffixes (exclamation and question marks).*/
function mapColumnName(columnName: string): string {
  const lastCharacter = columnName[columnName.length - 1];
  const isHintedColumn = lastCharacter === '!' || lastCharacter === '?';
  return isHintedColumn ? columnName.slice(0, -1) : columnName;
}
function mapQueryResultRows(rows: any[]): any[] {
  for (const row of rows) {
    for (const columnName in row) {
      if (isHintedColumn) {
        row[mapColumnName(columnName)] = row[columnName];
        delete row[columnName];
      }
    }
  }
  return rows;
}
```

Reply:
The only thing is that `isHintedColumn` isn't defined then in the `mapQueryResultRows` function, but I will fix it :).